File: core-csharp-practice/leet-code-codebase/SecondHighest.java
class SecondHighest {
    public static void main(String[] args) {
        int[] arr = {10, 20, 5, 40, 30};

        int highest = arr[0];
        int secondHighest = arr[0];

        for (int i = 1; i < arr.length; i++) {
            if (arr[i] > highest) {
                secondHighest = highest;
                highest = arr[i];
            } else if (arr[i] > secondHighest && arr[i] != highest) {
                secondHighest = arr[i];
            }
        }

        System.out.println("Second Highest number is: " + secondHighest);
    }
}

File: core-csharp-practice/leet-code-codebase/SecondHighest.cs

class SecondHighest
{
    public static void Main(string[] args)
    {
        int[] arr = { 10, 45, 22, 89, 34 };

        int highest = int.MinValue;
        int secondHighest = int.MinValue;

        foreach (int num in arr)
        {
            if (num > highest)
            {
                secondHighest = highest;
                highest = num;
            }
            else if (num > secondHighest && num != highest)
            {
                secondHighest = num;
            }
        }

        Console.WriteLine("Second Highest Number is: " + secondHighest);
    }
}
